storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -2132,33 +2135,30 @@ fseg_alloc_free_page_low(
 		/* fsp_alloc_free_page() invoked fsp_init_file_page()
 		already. */
 		return(block);
-		/*-----------------------------------------------------------*/
 	} else {
-		/* 7. We allocate a new extent and take its first page
-		======================================================*/
+		/* In worst case, try to allocate a new extent
+		and take its first page */
 		ret_descr = fseg_alloc_free_extent(seg_inode, iblock, &xdes,
 						   space, mtr, err);
-
 		if (!ret_descr) {
 			ut_ad(!has_done_reservation || *err);
 			return nullptr;
 		} else {
 			ret_page = xdes_get_offset(ret_descr);
 		}
 	}
-
+alloc_done:
 	if (ret_page == FIL_NULL) {
 		/* Page could not be allocated */
```

Comment:
The label should be placed *after* the `if` body, because each `goto alloc_done` has already ensured that `ret_page != FIL_NULL`.